src/app/info/WikiItems.tsx

```diff
@@ -94,13 +115,98 @@ const WikiItems: React.FC<WikiItemsProps> = ({ rows }) => {
         const indexToRemove: number = displayRows.findIndex(
             (sortedRow) => row.wiki_key === sortedRow.wiki_key
         );
-        setDisplayRows((DisplayRows) => {
-            const temp = DisplayRows.slice();
+        setDisplayRows((displayRows) => {
+            const temp = displayRows.slice();
             temp.splice(indexToRemove, 1);
             return temp;
         });
         return indexToRemove;
     };
+
+    const [isSwapMode, setIsSwapMode] = useState(false);
+
+    const swapRows = async (row1: DbWikiRow, upwards: boolean): Promise<void> => {
+        if (isSwapMode) {
+            return;
+        }
+        setIsSwapMode(true);
+
+        const rowIndex1 = displayRows.findIndex((row) => {
+            return row.wiki_key == row1.wiki_key;
+        });
+        const rowIndex2 = rowIndex1 + (upwards ? -1 : 1);
+
+        if (rowIndex2 < 0 || rowIndex2 >= displayRows.length) {
+            setIsSwapMode(false);
+            return;
+        }
+
+        const row2 = displayRows[rowIndex2];
+
+        const { error } = await supabase.rpc("swap_two_wiki_rows", {
+            key1: row1.wiki_key,
+            key2: row2.wiki_key,
+        });
+
+        const auditLog = {
+            action: `move a wiki item ${row1.row_order <= row2.row_order ? "down" : "up"}`,
+            wikiId: row1.wiki_key,
+            content: {
+                itemName: row1.title,
+                oldRow_order: row1.row_order,
+            },
+        } as const satisfies Partial<AuditLog>;
+
+        if (error) {
+            const logId = await logErrorReturnLogId(
+                "Error with supabase function swap_two_wiki_rows",
+                {
+                    error: error,
+                }
+            );
+            setErrorMessage(`Failed to swap rows. Log ID: ${logId}`);
+
+            void sendAuditLog({
+                ...auditLog,
+                wasSuccess: false,
+                logId: logId,
+            });
+            return;
+        } else {
+            void sendAuditLog({
+                ...auditLog,
+                wasSuccess: true,
+                content: { ...auditLog.content, newRow_order: row2.row_order },
+            });
+        }
+        setIsSwapMode(false);
+
+        reorderRows(row1, row2);
+    };
+
+    const reorderRows = (row1: DbWikiRow, row2: DbWikiRow): void => {
+        const wiki_keys = displayRows.map((displayRows) => displayRows.wiki_key);
+
```

Comment:
Can we rename this to make it clear that this is just doing it in the front end and to distinguish it from the db function?